source/fs.c

```diff
@@ -318,9 +319,9 @@ void initialise_fs() {
     This function is not thread-safe.
 */
 char *dirname(char *path) {
-    static char result[MAXPATHLEN];
-    strncpy(result, path, MAXPATHLEN - 1);
-    result[MAXPATHLEN - 1] = '\0';
+    static char result[MAXNAMLEN];
```

Comment:
Why define `MAXPATHLEN`, and then switch usages to `MAXNAMLEN`?  I haven't actually tried to build this new code...are we still using `MAXPATHLEN` for other things?